### ISSUE TITLE
EditorConfig support

### ISSUE THREAD

**Issue by exalted** [zed-industries/extensions#134]
[EditorConfig](https://editorconfig.org) is somewhat of an industry-standard these days. Although not a priority for me, you might want to take into consideration being compatible with existing EditorConfig configurations.

**Comment by BuonOmo** [comment]
@JosephTLyons do you plan on having rust interaction capable extensions ? TBH it still feels like it should be out of the box to me (as it is [in many places](https://editorconfig.org/#pre-installed)). This is a standard, and it would help a lot to have it not being an extension to ensure people are following project guidelines correctly!

**Comment by Moshyfawn** [comment]
> Also TBH it still feels like it should be out of the box to me (as it is [in many places](https://editorconfig.org/#pre-installed)).

The aim is to move the majority of languages to extensions, separating development and improvements from Zed. Then, pick multiple language extensions that can be used out of the box.

**Comment by JosephTLyons** [comment]
> @JosephTLyons do you plan on having rust interaction capable extensions ?
> 
> Also TBH it still feels like it should be out of the box to me (as it is [in many places](https://editorconfig.org/#pre-installed)). This is kind of a standard, and it would help a lot to have it not being an extension to ensure people are following project guidelines correctly!

It still could be a language we ship directly - we havent fully decided which languages will be ultimately shipped and which ones will be made into extensions - I just moved all language requests into this repo to clean up the issues in the Zed repo. We can always move things into Zed, if it makes sense.

**Comment by BuonOmo** [comment]
I think there is a confusion here. The support of the editor config language is not the most important. What matters most is that it is used by zed to properly format the code. Hence I'm not sure it is about the language more than it is about the integration into Zed's settings. 

I don't want to talk for everyone, but for me that's the most important part.

**Comment by bdougherty** [comment]
Yes, it's about supporting the editorconfig file for defining editor settings.

As far as the language goes, I'm pretty sure it is just standard `ini`.

**Comment by joeldrapper** [comment]
Yes, this issue is about making `.editorconfig` act like a `.zed/settings.json` file. Support for the INI language is, in my opinion, a much lower priority, and it’s not what this issue is about.

**Comment by jabr** [comment]
This is not a language extension. EditorConfig support is a basic feature of a modern code editor, up there with find-and-replace and configurable font settings.

**Comment by Moshyfawn** [comment]
I see what you're saying. The fact that it was moved to this extensions repo confused me. I'm not sure this can be made into a plugin if we're talking about Zed inheriting the `.editorconfig` config, but I might be wrong.
